main.tf

```
@@ -397,14 +397,14 @@ locals {
       regex       = "^[a-zA-Z0-9]+$"
     }
     cosmosdb_account = {
-      name        = substr(join("-", compact([local.prefix, "cosmos", local.suffix])), 0, 63)
-      name_unique = substr(join("-", compact([local.prefix, "cosmos", local.suffix_unique])), 0, 63)
+      name        = substr(join("-", compact([local.prefix, "cosmos", local.suffix])), 0, 44)
+      name_unique = substr(join("-", compact([local.prefix, "cosmos", local.suffix_unique])), 0, 44)
       dashes      = true
       slug        = "cosmos"
-      min_length  = 1
-      max_length  = 63
-      scope       = "resourceGroup"
-      regex       = "^[a-z0-9][a-z0-9-_.]+[a-z0-9]$"
+      min_length  = 3
+      max_length  = 44
+      scope       = "global"
+      regex       = "^[a-z0-9][a-z0-9-]{2,}$"
```

```suggestion
      regex       = "^[a-z0-9][a-z0-9-]{1,}[a-z0-9]$"
```

Even though the documentation doesn't say it, calling [Check Name Exists](https://learn.microsoft.com/en-us/rest/api/cosmos-db-resource-provider/database-accounts/check-name-exists?view=rest-cosmos-db-resource-provider-2025-10-15&tabs=HTTP) and [Create](https://learn.microsoft.com/en-us/rest/api/cosmos-db-resource-provider/database-accounts/create-or-update?view=rest-cosmos-db-resource-provider-2025-10-15&tabs=HTTP) with a name with a trailing hyphen `-` returns HTTP 400 (message: "The character '-' at index 12 is not allowed in the DatabaseAccount name").